pandas/io/sql.py

```diff
@@ -1248,19 +1248,23 @@ def to_sql(self, frame, name, if_exists='fail', index=True,
                          schema=schema, dtype=dtype)
         table.create()
         table.insert(chunksize)
-        # check for potentially case sensitivity issues (GH7815)
-        engine = self.connectable.engine
-        with self.connectable.connect() as conn:
-            table_names = engine.table_names(
-                schema=schema or self.meta.schema,
-                connection=conn,
-            )
-        if name not in table_names:
-            warnings.warn("The provided table name '{0}' is not found exactly "
-                          "as such in the database after writing the table, "
-                          "possibly due to case sensitivity issues. Consider "
-                          "using lower case table names.".format(name),
-                          UserWarning)
+        if (not name.isdigit() and not name.islower()):
+            # check for potentially case sensitivity issues (GH7815)
+            # Only check when name is not a number and name is not lower case
+            engine = self.connectable.engine
+            with self.connectable.connect() as conn:
+                table_names = engine.table_names(
```

Comment:
@jreback Apparently the warning itself can't be tested
https://github.com/pydata/pandas/blob/master/pandas/io/tests/test_sql.py#L919
I can add a test to ensure the warning is not triggered when using a number as a table name though?